### PR TITLE
Cleanup of relational storage code

### LINKED ISSUE #650: SQL system store improvements
As per @veikkoeeva [request on gitter](https://gitter.im/dotnet/orleans?at=55b2b57abab860d81689e9b6):

- [x]  Clean up comments in the SQL script
- [x]  Add documentation of SQL membership table [here](http://dotnet.github.io/orleans/Runtime-Implementation-Details/Cluster-Management) "2.SQL Server - TBD"
- [ ]  Consider checking in InitializeOrleansQueriesAsync if the queries were already loaded.

### PR DESCRIPTION
**Pull request by veikkoeeva**
Cleanup of relational storage code

* Removed dependency on statics when using relational storage.
* Use of constants clarified by grouping them by usage.
* Constant classes saved to each of their own files.
* Introduced SqlTestsEnvironment (by @kucheruk) also to Liveness test.
* Refactored the relational test initialization to make it easier to
  introduce other backends to the mix.
* Added example usage documentation on IRelationalStorage usage.
* Enhanced the multi-insert helper.

Still to do, batch insert could use a special "skip list" for columns
that will not be batch inserted and consequently simplifying Orleans
statistics inserts code, which is a near duplicate at the moment.

Fixes #650 when ready.

### CODE REVIEW

**Comment by veikkoeeva** [comment]
This is not ready yet, but I put it already here in case someone comes up with a question or comments.

File: src/OrleansProviders/SQLServer/CreateOrleansTables_SqlServer.sql [hunk 1]
```diff
@@ -105,11 +105,11 @@ GO
 -- These can be redefined (e.g. to provide non-destructive updates) provided the stated interface principles hold.
 CREATE TABLE [OrleansQuery]
 (	
-    [Key] VARCHAR(64) NOT NULL,
```
Comment (veikkoeeva) [inline src/OrleansProviders/SQLServer/CreateOrleansTables_SqlServer.sql]: This was originally an unfortunate choice as it is a reserved word in SQL Server and likely in other databases too.

**Comment by gabikliot** [comment]
@veikkoeeva , is it ready to be merged?

**Comment by veikkoeeva** [comment]
@gabikliot Not yet. Shouldn't be much going in. Sunday the latest, hopefully sooner.

**Comment by gabikliot** [comment]
No worries, no rush at all. Was just checking.

**Comment by veikkoeeva** [comment]
@gabikliot I think is ready now unless someone comes up with concerns.

**Pull request merged**